src/rust-crypto/RoomEncryptor.ts

```diff
@@ -142,19 +142,29 @@ export class RoomEncryptor {
         // not a particular problem, since `OlmMachine.updateTrackedUsers` just adds any users that weren't already tracked.
         if (!this.lazyLoadedMembersResolved) {
             await this.olmMachine.updateTrackedUsers(members.map((u) => new RustSdkCryptoJs.UserId(u.userId)));
-            this.lazyLoadedMembersResolved = true;
             logger.debug(`Updated tracked users`);
-        }
+            this.lazyLoadedMembersResolved = true;
 
-        // Query keys in case we don't have them for newly tracked members.
-        // This must be done before ensuring sessions. If not the devices of these users are not
-        // known yet and will not get the room key.
-        // We don't have API to only get the keys queries related to this member list, so we just
-        // process the pending requests from the olmMachine. (usually these are processed
-        // at the end of the sync, but we can't wait for that).
-        // XXX future improvement process only KeysQueryRequests for the tracked users.
-        logger.debug(`Processing outgoing requests`);
-        await this.outgoingRequestManager.doProcessOutgoingRequests();
+            // Query keys in case we don't have them for newly tracked members.
+            // It's important after loading members for the first time, as likely most of them won't be
+            // known yet and will be unable to decrypt messages despite being in the room for long.
+            // This must be done before ensuring sessions. If not the devices of these users are not
+            // known yet and will not get the room key.
+            // We don't have API to only get the keys queries related to this member list, so we just
+            // process the pending requests from the olmMachine. (usually these are processed
+            // at the end of the sync, but we can't wait for that).
+            // XXX future improvement process only KeysQueryRequests for the users that have never been queried.
+            logger.debug(`Processing outgoing requests`);
+            await this.outgoingRequestManager.doProcessOutgoingRequests();
+        } else {
+            // if members are already loaded it's less critical to await on key queries.
+            // We might still want to trigger a processOutgoingRequests here, if the
+            // query is done timely enough, the freshly tracked users will get the room key.
+            logger.debug(`Processing outgoing requests in background`);
+            this.outgoingRequestManager.doProcessOutgoingRequests().then(() => {
+                // nop
+            });
```

Comment:
this `then` doesn't seem to be doing anything?

Reply:
Just to signal explicitly that it's async and that we don't await (also removes a warning). What would you recommand for that case?

Reply:
TBH I'm surprised it removes the warning, though I do see the same thing in WebStorm.

I'd probably just add a comment in this case. Another option is it add a `catch` block which logs any exceptions that get thrown.